esp/esp/program/views.py

```diff
@@ -524,11 +524,14 @@ def newprogram(request):
             perms, modules = prepare_program(temp_prog, form.cleaned_data)
             #   Save the form's raw data instead of the form itself, or its clean data.
             #   Unpacking of the data happens at the next step.
+            pretty_modules = []
+            for module in modules:
+                pretty_modules.append(module[0].replace("Program Module: ",""))
```

Comment:
Thanks, this is certainly a bit better.  You could make it even a bit cleaner, though: instead of doing it here, push it one step further back into `prepare_program`, where we generate `modules` in the first place.  You'll see that there we do `str(<ProgramModule object>)`.

Now, you could just replace that with `str(...).replace(...)`, but if you look at how `ProgramModule` computes `str` (see its `__unicode__` method), you'll see it's just `'Program Module: %s' % self.admin_title`.  So even better, you can just replace the `str()` with `.admin_title` and call it a day!  That way, the code will be simpler, and someone else modifying it is less likely to break your string-replace because they change something else.

Reply:
Thanks, that's a good idea